R/fair-aaa.R

```diff
@@ -0,0 +1,221 @@
+#' Create group-wise metrics
+#'
+#' Group-wise metrics quantify the disparity in value of a metric across a
+#' number of groups. Group-wise metrics with a value of zero indicate that the
+#' underlying metric is equal across groups. yardstick defines
+#' several common fairness metrics using this function, such as
+#' [demographic_parity()], [equal_opportunity()], and [equalized_odds()].
+#'
+#' Note that _all_ yardstick metrics are group-aware in that, when passed
+#' grouped data, they will return metric values calculated for each group.
+#' When passed grouped data, group-wise metrics also return metric values
+#' for each group, but those metric values are calculated by first additionally
+#' grouping by the variable passed to `by` and then summarizing the per-group
+#' metric estimates across groups using the function passed as the
+#' `aggregrate` argument.
+#'
+#' @param fn A yardstick metric function or metric set.
+#' @param name The name of the metric to place in the `.metric` column
+#' of the output.
+#' @param aggregrate A function to summarize the generated metric set results
+#' `x`. In many cases, `~diff(range(x$.estimate))` or
+#' `~r <- range(x$.estimate); r[1]/r[2]`.
```

Comment:
This documentation should be expanded. I would look at `?purrr::map()` for inspiration.

Right now, the examples listed are technically not functions

Reply:
Thanks for bringing this up—addressed in [4436785](https://github.com/tidymodels/yardstick/pull/434/commits/4436785759021d88db4360ce7f6f62b2d419f69f). Opted to describe the input and output type more thoroughly and refer to other sections for worked examples.